src/peergos/shared/user/fs/FileWrapper.java

```diff
@@ -654,24 +654,32 @@ public static int getNumberOfChunks(long size) {
                                                     throw new IllegalStateException("File already exists with name " + filename);
                                                 FileWrapper child = childOpt.get();
                                                 FileProperties childProps = child.getFileProperties();
+
+                                                TriFunction<FileWrapper, Snapshot, AsyncReader, CompletableFuture<Snapshot>> recalculateThumbnailIfNecessary =
+                                                        (updatedChild, latestSnapshot, is) -> {
+                                                    if(startIndex != 0) {
```

Comment:
you could use an inline statement in both places. arguable whether or not that is simpler/shorter
e.g. startIndex != 0 ? Futures.of(latestSnapshot) : updatedChild.recalculateThumbnail(...)